apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,168 @@
+	" [Operation] : 1.Read/Write / 2.Read / 3.Write\n"						\
+	" [Log] : Log Print Option [y/n]\n"										\
+	"\n ex) fs_rw_test 128 30 1 y\n"
+
+/****************************************************************************
+ * fs_rw_test
+ ****************************************************************************/
+
+static inline long get_time(void)
+{
+	struct timeval tv;
+	gettimeofday(&tv, NULL);
+	return (tv.tv_sec * 1000 + tv.tv_usec / 1000);
+}
+
+static void rw_testlog(char *printopt, const char *format, ...)
+{
+	va_list ap;
+	if (printopt[0] == 'y' || printopt[0] == 'Y') {
+		va_start(ap, format);
+		vfprintf(stdout, format, ap);
+		va_end(ap);
+	}
+}
+
+static FILE *t_fopen(const char *path, const char *mode, char *printopt)
+{
+	FILE *fp;
+	long t, dt;
+	t = get_time();	
+	fp = fopen(path, mode);
+	if (fp == NULL) {
+		printf("fopen Failed\n");
+		return NULL;
+	}
+	dt = get_time() - t;
+	rw_testlog(printopt, "fopen(%c):%dms, ", mode[0], dt);
+	return fp;	
+}
+
+static void t_fwrite(const void *data, size_t size, size_t n_items, FILE *stream, char *printopt)
+{
+	long t, dt;
+	t = get_time();
+	fwrite(data, size, n_items, stream);
+	dt = get_time() - t;
+	rw_testlog(printopt, "fwrite:%dms, ", dt);
+}
+
+static void t_fclose(FILE *stream, char *printopt)
+{
+	long t, dt;
+	t = get_time();
+	fclose(stream);
+	dt = get_time() - t;
+	rw_testlog(printopt, "fclose:%dms\n", dt);
+}
+
+static void t_fread(void *ptr, size_t size, size_t n_items, FILE *stream, char *printopt)
+{
+	long t, dt;
+	char *data = ptr;
+	t = get_time();		
+	fread(ptr, size, n_items, stream);
+	dt = get_time() - t;
+	rw_testlog(printopt, "fread:%dms, data:%d ", dt, (int)data[0]);
+}
+
+void rw_operation(int size, int operation, int iteration, char *printopt)
+{
+	FILE *fp = NULL;
+	char data[size];
+	char buf[size];
+
+	rw_testlog(printopt, "==========iteration %d==========\n", iteration);
+	switch (operation) {
+	case 1:
+		fp = t_fopen(TEST_FILE_PATH TEST_FILE_NAME, "w", printopt);
+		data[0] = iteration;
+		t_fwrite(data, sizeof(char), sizeof(data), fp, printopt);
+		t_fclose(fp, printopt);
+		fp = t_fopen(TEST_FILE_PATH TEST_FILE_NAME, "r", printopt);
+		t_fread(buf, sizeof(char), sizeof(buf), fp, printopt);
+		break;
+	case 2:
+		fp = t_fopen(TEST_FILE_PATH TEST_FILE_NAME, "r", printopt);
+		t_fread(buf, sizeof(char), sizeof(buf), fp, printopt);
+		break;
+	case 3:
+		fp = t_fopen(TEST_FILE_PATH TEST_FILE_NAME, "w", printopt);
+		memset(data, iteration, sizeof(data));
+		t_fwrite(data, sizeof(char), sizeof(data), fp, printopt);
+		break;
+	}
+	t_fclose(fp, printopt);
+	
+	rw_testlog(printopt, "================================\n");
+}
+
+void f_rw_performance(char *path, int size, int iteration, int operation, char *printopt)
+{
+	long t, dt;
+
+	t = get_time();	
+	for (int i = 0; i < iteration; i++) {
+		rw_operation(size, operation, i, printopt);
+	}
+	dt = get_time() - t;
+	rw_testlog("y", "get_time : %d ms\n", dt);
```

Comment:
I don't understand why you use new API, rw_testlog to print.
1. The decision of printing is already hard-coded. User can not control it on run-time.
2. Why do you deliver a character, "y"? why not just integer value?

Reply:
1 . rw_testlog
This part does not need rw_testlog, so I will change it.
But the rest is a separate function that allows the user to choose whether to print the log.

In the filesystem r/w, the log output should be disabled because it can affect speed. Conversely, if you want to see detailed speed and data write success, you should be able to activate it.

2. why deliver "y"?
y/n When we express two cases, we judged y / n to be intuitive rather than 1,2